### ISSUE TITLE
Add an option to support namespaced expect

### ISSUE THREAD

**Issue by StephanBijzitter** [jest-community/eslint-plugin-jest#286]
Hello!

I'm not using Jest, but I am using Expect (which is used by Jest) as part of a custom testing suite with Puppeteer.

This plugin works absolutely fine when I do this:
```ts
expect(percentage).toStrictEqual('10%');
```
But... `expect` is not globally available for me, so my code actually looks like this:
```ts
page.expect(percentage).toStrictEqual('10%');
```

Is there some way I can let this plugin work the way I need it to? Otherwise, I'd love to request this as a feature and if pointed in the right direction, I may be able to get a pull request going myself.

**Comment by SimenB** [comment]
what about

```js
const {expect} = page;

expect(percentage).toStrictEqual('10%');
```
?

If that does _not_ work we could explore an option. But accessing properties instead of "top level" function calls is a bit more work in the rules. So preferably that's avoidable

**Comment by StephanBijzitter** [comment]
Nice one :-)

I experimented with this yesterday too, but instead opted for something else, because it would be very difficult to make sure other programmers (and testers) will do it that way.

So I figured I'd expose `expect` in a different way:

```ts
// Clarification: `page` is not a puppeteer Page object,
// even though it looks like it a lot, it can also be Selenium-based

await page.assert('the shown percentage is 20%', async (expect) => {
    const percentage = await page.getText('#redacted');
    expect(percentage).toStrictEqual('20%');
});
```

This actually works out better than what I originally requested anyway, as it allows us to log more beautiful messages to Slack/Github/Whatever from our system.

So yeah, both your solution and the one above would work for us, so this feature request has no real value to me personally anymore.
